api/diagnostics_test.go

```diff
@@ -426,44 +423,24 @@ func TestRunSnapshot(t *testing.T) {
 	assert.Equal(t, responseJSON.Status, "Job has been successfully started")
 	assert.NotEmpty(t, responseJSON.Extra.LastBundleFile)
 
-	assert.True(t, waitForBundle(t, router))
-
-	snapshotFiles, err := ioutil.ReadDir(dt.Cfg.FlagDiagnosticsBundleDir)
-	assert.NoError(t, err)
-	assert.True(t, len(snapshotFiles) > 0)
-
-	stat, err := os.Stat(bundle)
-	assert.NoError(t, err)
-	assert.False(t, stat.IsDir())
+	bundle := filepath.Join(dt.Cfg.FlagDiagnosticsBundleDir, responseJSON.Extra.LastBundleFile)
+	assert.True(t, waitForBundle(t, bundle))
 }
 
-func waitForBundle(t *testing.T, router *mux.Router) bool {
+func waitForBundle(t *testing.T, bundlePath string) bool {
 	timeout := time.After(2 * time.Second)
 	for {
 		select {
 		case <-timeout:
 			t.Log("Timeout!")
 			return false
 		default:
-			response, code, err := MakeHTTPRequest(t, router, "http://127.0.0.1:1050/system/health/v1/report/diagnostics/status", "GET", nil)
+			stat, err := os.Stat(bundlePath)
```

Comment:
Even though it is a lighter operation than what it used to be, in these situations I'd add a small sleep after this instead of looping as fast as possible while polling for the file.